fix name comparison

Bia10 opened the pull request.

Bia10 committed:
fix name comparison

Kaioru commented:
The original code is kept that way to ignore string casing when checking for duplicate ID's. Although (probably) more optimised, the proposed change removes that functionality.

Bia10 commented:
The original code leads to 

System.NotSupportedException: Method Equals(string,StringComparison) in String are not supported when convert to BsonExpression (c.Name.Equals(value(Edelstein.Service.Login.Handlers.CheckDuplicatedIDHandler+<>c__DisplayClass0_0).name, InvariantCultureIgnoreCase)).

Kaioru commented:
In that case, something like `c.Name.EqualsIgnoreCase(name)` would do the trick instead of a `==` condition. As the code in this PR leads to characters being able to have the same name with different casings.

Though, there might be other ways of doing this.. Feel free to suggest!

Bia10 committed:
Merge pull request #1 from Kaioru/rewrite

merge

Bia10 committed:
Merge pull request #2 from Kaioru/rewrite

merge

Bia10 committed:
Merge pull request #3 from Kaioru/rewrite

Merge

Bia10 committed:
remove unused directives
simplify using statements

Bia10 committed:
fix namespace mismatch
remove empty constructor

Bia10 committed:
solution update to version 16
project update to net.core 3.1
dependencies update to latest (excluding Foundatio)
fix missing brace in ModifyStats method
fix namespace mismatch

Bia10 committed:
remove extra space introduced during refactor
